src/shaders/fill_extrusion_pattern.vertex.glsl

```diff
@@ -130,16 +140,20 @@ void main() {
     y_ground += y_ground * 5.0 / max(3.0, top_height);
 #endif
     v_ao = vec3(mix(concave, -concave, start), y_ground, h - ele);
-    directional *= (1.0 + 0.05 * (1.0 - top_up_ny.y) * u_ao[0]); // compensate sides faux ao shading contribution
+    NdotL *= (1.0 + 0.05 * (1.0 - top_up_ny.y) * u_ao[0]); // compensate sides faux ao shading contribution
 
 #ifdef PROJECTION_GLOBE_VIEW
     top_height += u_height_lift;
 #endif
     gl_Position.z -= (0.0000006 * (min(top_height, 500.) + 2.0 * min(base, 500.0) + 60.0 * concave + 3.0 * start)) * gl_Position.w;
 #endif
 
-    v_lighting.rgb += clamp(directional * u_lightcolor, mix(vec3(0.0), vec3(0.3), 1.0 - u_lightcolor), vec3(1.0));
+#ifdef LIGHTING_3D_MODE
+    v_lighting.rgb += NdotL;
```

Comment:
what does this mean?

Reply:
I am basically only passing NdotL when `LIGHTING_3D_MODE` is enabled. `v_lighting` is the varying which only contains the light contribution in this shader. The albedo colors are determined in the fragment shader.